pkg/schedule/schedulers/hot_region_v2.go

```diff
@@ -58,19 +84,29 @@ func newRankV2Ratios(balancedRatio, perceivedRatio, minHotRatio float64) *rankV2
 		balancedRatio = 0.95
 	}
 
-	rs := &rankV2Ratios{balancedRatio: balancedRatio, perceivedRatio: perceivedRatio, minHotRatio: minHotRatio}
-	// preBalancedRatio = 1.0 - 2*(1.0-balancedRatio)
-	// The maximum value with `balancedRatio-0.1` is to prevent the preBalance range becoming too large.
-	rs.preBalancedRatio = math.Max(2.0*balancedRatio-1.0, balancedRatio-0.1)
-	rs.balancedCheckRatio = balancedRatio - 0.02
-	rs.preBalancedCheckRatio = rs.preBalancedRatio - 0.03
+	futureStateChecker := &balanceChecker{
+		balancedRatio: balancedRatio,
+		// preBalancedRatio = 1.0 - 2*(1.0-balancedRatio)
+		// The maximum value with `balancedRatio-0.1` is to prevent the preBalance range becoming too large.
+		preBalancedRatio: math.Max(2.0*balancedRatio-1.0, balancedRatio-0.1),
+	}
+	currentStateChecker := &balanceChecker{
+		balancedRatio:    balancedRatio - 0.02,
+		preBalancedRatio: futureStateChecker.preBalancedRatio - 0.03,
```

Comment:
Do you mean minus `firstPriorityMinHotRatio` and `secondPriorityMinHotRatio` there?

Reply:
no, only make futureChecker is stricter than currentChecker

Reply:
Do we need to extract them as const?